posthog/api/documentation.py

```diff
@@ -39,11 +39,17 @@ def get_security_requirement(self, auto_schema):
             if isinstance(permission, APIScopePermission):
                 try:
                     scopes = permission._get_required_scopes(request, view)
-                    return [{self.name: scopes}]
+                    if scopes:
+                        return [{self.name: scopes}]
+                    else:
+                        return []
```

Comment:
Nitpick: Can reduce nesting (and PR diff) by simply adding an early return only.

```suggestion
                    if not scopes:
                        return []
                    return [{self.name: scopes}]
```